walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -25,14 +25,47 @@
   /**
    * Queries a collection of available time ranges that could accommodate the meeting request
    * @param {Collection<Event>} events The collection of events that may conflict with the reqest
-   * @param {MeetingRequest} request The request to query abailable time ranges of
-   * @return {Collection<TimeRange>} The time ranges that could accommodate the meeting request
+   * @param {MeetingRequest} request The request to query available time ranges of
+   * @return Collection<TimeRange> The time ranges that could accommodate the meeting request
    */
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
+    Collection<String> attendeesWithOptional = new ArrayList<String>();
+    Collection<String> attendeesWithoutOptional = new ArrayList<String>();
+    
+    // Creates two collections of attendees for only mandatory and with optional
+    for (String attendee : request.getAttendees()) {
+      attendeesWithOptional.add(attendee);
+      attendeesWithoutOptional.add(attendee);
+    }
+    for (String optional : request.getOptionalAttendees()) {
+      attendeesWithOptional.add(optional);
+    }
+
+    // Queries the time ranges for both collections
+    Collection<TimeRange> rangesWithOptional = queryAllAttendees(events, request, attendeesWithOptional);
+    Collection<TimeRange> rangesWithoutOptional = queryAllAttendees(events, request, attendeesWithoutOptional);
+
+    // Use optional time ranges if there are no mandatory attendees
+    if (request.getAttendees().size() == 0) {
```

Comment:
Let's use request.getAttendees().isEmpty() instead. BTW, will request.getAttendees() returns a null as the result?

Reply:
changed with most recent commit

Reply:
and in the MeetingRequest class, the attendees array list is initialized to an empty collection, so I don't believe a null result is possible.